recipes/install_database.rb

```diff
@@ -17,33 +17,69 @@
 
 Chef::Recipe.send(:include, Abiquo::Commands)
 
-mysqlcmd = mysql_cmd(node['abiquo']['db'])
+mysql2_chef_gem 'server' do
+    provider Chef::Provider::Mysql2ChefGem::Mariadb
+    action :install
+end
+
+# Workaround
+# https://github.com/chef-cookbooks/database/issues/151
+ruby_block "modifying MySQL2 client" do
+    block do
+        Mysql2::Client.default_query_options[:connect_flags] |= Mysql2::Client::MULTI_STATEMENTS
+    end
+    action :run
+end
 
-execute "create-database" do
-    command "#{mysqlcmd} -e 'CREATE DATABASE kinton'"
-    not_if "#{mysqlcmd} kinton -e 'SELECT 1'"
-    notifies :run, "execute[install-database]", :immediately
+conn_info = {
+    :host     => node['abiquo']['db']['host'],
+    :username => node['abiquo']['db']['user'],
+    :password => node['abiquo']['db']['password'],
+    :port     => node['abiquo']['db']['port']
+}
+
+# Create DB
+mysql_database 'kinton' do
+    connection conn_info
+    action :create
 end
 
-execute "install-database" do
-    command "#{mysqlcmd} kinton </usr/share/doc/abiquo-server/database/kinton-schema.sql"
-    action :nothing
-    notifies :run, "ruby_block[extract-m-user-password]", :immediately
-    notifies :run, "execute[install-license]", :immediately
+# Populate schema
+mysql_database 'kinton-schema' do
+    connection conn_info
+    database_name 'kinton'
+    sql { ::File.open('/usr/share/doc/abiquo-server/database/kinton-schema.sql').read }
+    action :query
+    not_if do
+        client = Mysql2::Client.new(conn_info.merge(:database => 'kinton'))
+        query = "select count(*) as count from information_schema.tables where table_name = 'DATABASECHANGELOG' and table_schema = 'kinton'"
+        client.query(query).first['count'] > 0
+    end
 end
 
-execute "install-license" do
-    command "#{mysqlcmd} kinton -e \"INSERT INTO license (data) VALUES ('#{node['abiquo']['license']}');\""
-    action :nothing
+# Install license if present
+mysql_database 'install-license' do
+    connection conn_info
+    database_name 'kinton'
+    sql "INSERT INTO license (data) VALUES ('#{node['abiquo']['license']}')"
+    action :query
     not_if { node['abiquo']['license'].nil? || node['abiquo']['license'].empty? }
 end
 
+# Extract M user password from databases
+# Randomly generated after liquibase run
 ruby_block "extract-m-user-password" do
     block do
-        Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)      
-        mysql_command = "#{mysqlcmd} kinton -B --skip-column-names -e \"select COMMENTS from DATABASECHANGELOG where ID = 'default_user_for_m'\""
-        mysql_command_out = shell_out!(mysql_command)
-        node.set['abiquo']['properties']['abiquo.m.credential'] = mysql_command_out.stdout.gsub("\n", "")
+        client = Mysql2::Client.new(conn_info.merge(:database => 'kinton'))
+        loop do
+            puts "Waiting for all tables to be present..."
```

Comment:
This is also used before... Should we better move this to a library method so it gets properly encapsulated and is reusable?